x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1929,12 +2070,20 @@ private void maybeStartTokenRemover() {
         }
     }
 
-    String prependVersionAndEncodeAccessToken(TransportVersion version, String accessToken) throws IOException, GeneralSecurityException {
-        if (version.onOrAfter(VERSION_ACCESS_TOKENS_AS_UUIDS)) {
+    public String prependVersionAndEncodeAccessToken(TransportVersion version, byte[] accessTokenBytes) throws IOException,
+        GeneralSecurityException {
+        if (version.onOrAfter(VERSION_GET_TOKEN_DOC_FOR_REFRESH)) {
+            try (BytesStreamOutput out = new BytesStreamOutput()) {
```

Comment:
nit: We know how much bytes we will write. I suggest we pass in the expected size to the `BytesStreamOutput` constructor in order to avoid relocation of the underlying byte array.

Reply:
Also handled in https://github.com/elastic/elasticsearch/pull/97395/commits/2a2e7f75e6a1fc6d3ce09fca80ffc89f614937d0 .